### PR TITLE
Demisto-SDK docker

### PR DESCRIPTION
**Pull request by jochman**
## Status
Ready

## Description
A docker image with demisto-sdk for local run/ci run for customers.

### CODE REVIEW

File: docker/demisto-sdk/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+(All examples from the [content](https://github.com/demisto/content) repository).
+
+### Validate command
+
+```sh
+docker run -it --rm --mount type=bind,source="$(pwd)",target=/content demisto/demisto-sdk:<tag> demisto-sdk validate -i Packs/ipinfo/Integrations/ipinfo_v2
+```
+
+#### Break down the command
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:28]:
```suggestion
#### Breaking down command arguments
```

File: docker/demisto-sdk/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+- docker run  
+    Creates a container and run a command
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:31]:
```suggestion
    Creates a container (if one does not exist) and runs the following command inside it
```

File: docker/demisto-sdk/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+- -it  
+    Keep the stdin open and tty
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:33]:
```suggestion
    Keeps the `stdin` open and tty
```

`and tty`?

File: docker/demisto-sdk/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+- --rm  
+    Removes the docker container when done (Remove if you want to reuse the container)
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:35]:
```suggestion
    Removes the docker container when done (ommit this part to re-use the container in the future)
```

File: docker/demisto-sdk/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+- --mount type=bind,source="$(pwd)",target=/content  
+    Connect the pwd (assuming you're in content) to the content directory in the container
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:37]:
```suggestion
    Connects the pwd (assuming you're in content) to the container's content directory
```

File: docker/demisto-sdk/README.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+- demisto/demisto-sdk:\<tag> (Replace the tag with locked version)
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:38]: where does one get the tag?

File: docker/demisto-sdk/README.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+- docker run  
+    Creates a container and run a command
+- -it  
+    Keep the stdin open and tty
+- --rm  
+    Removes the docker container when done (Remove if you want to reuse the container)
+- --mount type=bind,source="$(pwd)",target=/content  
+    Connect the pwd (assuming you're in content) to the content directory in the container
+- demisto/demisto-sdk:\<tag> (Replace the tag with locked version)  
+    The docker image name
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:39]: huh?

File: docker/demisto-sdk/README.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+- -it  
+    Keep the stdin open and tty
+- --rm  
+    Removes the docker container when done (Remove if you want to reuse the container)
+- --mount type=bind,source="$(pwd)",target=/content  
+    Connect the pwd (assuming you're in content) to the content directory in the container
+- demisto/demisto-sdk:\<tag> (Replace the tag with locked version)  
+    The docker image name  
+- demisto-sdk validate -i Packs/ipinfo/Integrations/ipinfo_v2
+    The command to run
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:41]:
```suggestion
    The demisto-sdk command to be run inside the container
```

File: docker/demisto-sdk/README.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+### Lint command
+
+If you want to run the `lint` command, you must connect the docker daemon (docker in docker)
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:45]:
```suggestion
To run the `lint` command, connect the docker daemon (docker inside a docker), as lint itself is run inside a docker container. 
```

File: docker/demisto-sdk/README.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+```sh
+docker run -it --rm --mount type=bind,source="$(pwd)",target=/content --mount source=/var/run/docker.sock,target=/var/run/docker.sock,type=bind jochman/demisto-sdk <demisto-sdk-command>
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:48]: is it still `jochman/demisto-sdk`?

File: docker/demisto-sdk/README.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+#### The mount park
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:51]: park?

File: docker/demisto-sdk/README.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+- --mount source=/var/run/docker.sock,target=/var/run/docker.sock,type=bind  
+    Mount the docker daemon in container to the outside (docker in docker)
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:54]:
```suggestion
    Mounts the docker deamon container, to allow using docker commands from within a docker container.
```

File: docker/demisto-sdk/README.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
+# Demisto-SDK docker
+
+Run demisto-sdk commands from within a docker.
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:3]:
```suggestion
Allows running demisto-sdk commands from within a docker container.
```

File: docker/demisto-sdk/README.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+
+You can use this image to run validations locally on your work environment or inside your ci.
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:5]:
```suggestion
You can use this image to run demisto-sdk commands locally, or as part of a CI/CD process.
```

Comment (dorschw) [inline docker/demisto-sdk/README.md:5]:
```suggestion
You can use this image to run demisto-sdk commands locally, or as part of a CI/CD process.
```

File: docker/demisto-sdk/README.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+
+## The Content Repository
+
+To use the Demisto-SDK you should have a content-like repository (or use the official [content repository](https://github.com/demisto/content)).
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:9]:
```suggestion
To use the Demisto-SDK, make sure you have a content-like repository with Cortex XSOAR content, in a structure that matches the official [XSOAR Content repo](https://github.com/demisto/content).
 
 ```

File: docker/demisto-sdk/README.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+You can generate one from this [template](https://github.com/demisto/content-external-template)
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:10]:
```suggestion
Such a repository may be generated using the following [template](https://github.com/demisto/content-external-template)
```

File: docker/demisto-sdk/README.md [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+
+## Mounts
+
+We use volume mount in our environment to run on local content repository.
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:14]:
```suggestion
Demisto-sdk uses volume mounts, to run on local content repository.
```

end of the sentence is unclear

File: docker/demisto-sdk/README.md [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+Please note that there's a performance issue with mount on MacOS and Windows.
+You can solve the performance issue by running on Linux/Windows WSL2 or by cloning your repository to the `/content` directory inside the container.
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:16]:
```suggestion
Please note that mounting on MacOS and Windows may cause slowness. 

To ensure the best performance, please either:
- Use a linux machine
- Use [Windows WSL2](https://docs.microsoft.com/en-us/windows/wsl/install) 
- Clone your content repository to the `/content` directory inside the container.
```

do we explain how to do the cloning?

File: docker/demisto-sdk/README.md [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+
+## Examples
+
+(All examples from the [content](https://github.com/demisto/content) repository).
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:20]:
```suggestion
(All examples use Cortex XSOAR's official [content repository](https://github.com/demisto/content)).
```

File: docker/demisto-sdk/README.md [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+
+### Validate command
+
```
Comment (dorschw) [inline docker/demisto-sdk/README.md:23]:
```suggestion
for more information about the validate command, please refer to its [documentation](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/validate/README.md) on the [demisto-sdk repo](https://github.com/demisto/demisto-sdk).
```

**Review by dorschw — CHANGES_REQUESTED** [review]

File: docker/demisto-sdk/README.md (hunk 18 above)
Comment (jochman) [reply docker/demisto-sdk/README.md:17]: If someone is working with a docker, he should be familiar with vcs, container etc. I think we should draw a line between where we support and where we are not.

File: docker/demisto-sdk/README.md (hunk 17 above)
Comment (jochman) [reply docker/demisto-sdk/README.md]: It's not the demisto-sdk that uses the volume mounts.

File: docker/demisto-sdk/README.md (hunk 3 above)
Comment (jochman) [reply docker/demisto-sdk/README.md:34]: https://en.wikipedia.org/wiki/Tty_(Unix)

File: docker/demisto-sdk/README.md (hunk 11 above)
Comment (jochman) [reply docker/demisto-sdk/README.md:52]: part? :)

File: docker/demisto-sdk/README.md (hunk 10 above)
Comment (jochman) [reply docker/demisto-sdk/README.md:49]: nope, should update it as well.

File: docker/demisto-sdk/README.md (hunk 6 above)
Comment (jochman) [reply docker/demisto-sdk/README.md:39]: docker hub

File: docker/demisto-sdk/README.md (hunk 7 above)
Comment (jochman) [reply docker/demisto-sdk/README.md:40]: what is the line above

File: docker/demisto-sdk/README.md [hunk 21]
```diff
@@ -0,0 +1,59 @@
+# Demisto-SDK docker
```
Comment (Itay4) [inline docker/demisto-sdk/README.md:1]: we should document the env vars that need to be modified on `docker run` for some of the commands

**Review by Itay4 — COMMENTED** [review]
how does it work with lint? dont we need dind?

File: docker/demisto-sdk/README.md [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+- Clone your content repository to the `/content` directory inside the container.
+
+## Examples
+
+(All examples use Cortex XSOAR's official [content repository](https://github.com/demisto/content)).
+
+### Validate command
+for more information about the validate command, please refer to its [documentation](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/validate/README.md) on the [demisto-sdk repo](https://github.com/demisto/demisto-sdk).
+```sh
+docker run -it --rm --mount type=bind,source="$(pwd)",target=/content demisto/demisto-sdk:<tag> demisto-sdk validate -i Packs/ipinfo/Integrations/ipinfo_v2
```
[the remark this answers is not in the document]
Comment (jochman) [reply docker/demisto-sdk/README.md:30]: I think that it should be up to the customer. :)

[the remark this answers is not in the document]
Comment (dorschw) [reply docker/demisto-sdk/README.md]: will a read-only bind allow `format` calls?

**Review by dorschw — APPROVED** [review]

**Review by glicht — APPROVED** [review]

**Pull request merged**